app/assets/javascripts/mediation_table.js

```diff
@@ -0,0 +1,74 @@
+var mediationTable = (function() {
+  var defaultOptions = {
+    selector: '[data-mediate-request]'
+  };
+
+  return {
+    init: function(opts) {
+      var _this = this;
+      _this.options = $.extend(defaultOptions, opts);
+      $(document).on('ready page:load', function(){
+        _this.addToggleHoldingsBehavior();
+      });
+      return this;
+    },
+
+    options: {},
+
+    addToggleHoldingsBehavior: function() {
+      var _this = this;
+      _this.mediatableRows().each(function(){
+        _this.addToggleClick($(this));
+      });
+    },
+
+    mediatableRows: function() {
+      return $(this.options.selector);
+    },
+
+    addToggleClick: function(row) {
+      var _this = this;
+      var toggle = _this.toggleHandle(row);
+      toggle.on('click', function() {
+        if(_this.rowIsProcessed(row)) {
```

Comment:
Is there a way to made `addHoldings` short-circuit if the holdings already exist so this could just be e.g.

`toggle.on('click', function() { _this.showHoldings(row); } )`